corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -381,7 +478,13 @@ def get_payload(self):
 
     def get_response(self):
         try:
-            return HttpResponse(self.get_payload(), mimetype="text/xml")
+            payload = self.get_payload()
+            try:
+                f = open(payload, 'r')
+                # Since payload file is all one line, need to readline based on bytes
+                return StreamingHttpResponse(iter(lambda: f.readline(MAX_BYTES), ''), mimetype="text/xml")
+            except IOError:
```

Comment:
Unclear. this is actually a problem whereever I use that idiom. another solution could be to check if the payload has a `.xml` file extension since all the filenames should have that and none of the XML payload files should. That way we can still guard against an actual `IOError` in which case we might send back a 500 error with an appropriate error message